django/db/models/base.py

```diff
@@ -2105,6 +2105,7 @@ def _check_constraints(cls, databases):
                 # JOIN must happen at the first lookup.
                 first_lookup = lookups[0]
                 if (
+                    hasattr(field, 'get_transform') and
```

Comment:
I think we should rather take into account `one_to_one` fields in [L2101](https://github.com/django/django/pull/14255/files#diff-1c8b882c73bfda668d5451d4578c97191b0ebc0f088d0c0ba2296ab89b428c44R2101) :thinking: